storage_test.go

```diff
@@ -0,0 +1,442 @@
+package dynamodbstorage
+
+import (
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/caddyserver/certmagic"
+)
+
+const TestTableName = "CertMagicTest"
+const DisableSSL = true
+
+func initDb() error {
+	storage := Storage{
+		Table:         TestTableName,
+		AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+		AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+		AwsDisableSSL: true,
+	}
+	sess, err := session.NewSession(&aws.Config{
+		Endpoint:   &storage.AwsEndpoint,
+		Region:     &storage.AwsRegion,
+		DisableSSL: &storage.AwsDisableSSL,
+	})
+	if err != nil {
+		return err
+	}
+
+	svc := dynamodb.New(sess)
+
+	// attempt to delete table in case already exists
+	deleteTable := &dynamodb.DeleteTableInput{
+		TableName: aws.String(storage.Table),
+	}
+	_, err = svc.DeleteTable(deleteTable)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case dynamodb.ErrCodeResourceInUseException:
```

Comment:
Since default returns the same `aerr` as three others, is it worth keeping all of them.  If so, another options is to group the ones that return `aerr` on one line with commas as separators.

Reply:
true, that was sample code from AWS but i really don't need it in the tests like this, I'll clear that out